vrs/marten-construction-script.py

```diff
@@ -0,0 +1,364 @@
+    )
+    logger.info("Annotation finished")
+
+    # Create Table with Reference, Indel, Long Reference, Long Variant, and Sex Chromosome variants
+    hts = []
+    for variant_type in freq_dict:
+        variant_freq = freq_dict[variant_type]
+        desired_count = desired_counts[variant_type]
+        sampling_freq = desired_count / (whole_count * variant_freq)
+
+        ht_var = ht_whole.filter(ht_whole[variant_type])
+        ht_var = ht_var.sample(sampling_freq)
+        ht_var = ht_var.checkpoint(
+            f"{args.tmp_dir}/{variant_type}_temp.ht",
+            _read_if_exists=args.read_if_exists,
+            overwrite=args.overwrite,
+        )
+        hts.append(ht_var)
+        logger.info(f"Finished creating table for %s", variant_type)
+
+    ht_union = hts[0].union(*hts[1:])
+
+    logger.info(
+        f"Total size of random, indel, long ref, long var, sex chr combined Table as: %i",
+        ht_union.count(),
+    )
+    ht_union = ht_union.checkpoint(
+        f"{args.tmp_dir}/union_path_temp.ht",
+        _read_if_exists=args.read_if_exists,
+        overwrite=args.overwrite,
+    )
+
+    # Construct a Table of only multiallelic sites
+
+    # For 3 random partitions, only take the highly multiallelic sites (>5 variants per loci)
+    # NOTE: Construct Table only by partitions, and each partition has ~3200 multiallelic variants
+    # 3 partitions to yield 9600 multiallelic variants (default, closest to 10,000)
+    multiallelic_partitions = 3
+
+    if args.custom:
+        multiallelic_partitions = int(counts_list[-2]) // 3200
+        if multiallelic_partitions < 1 and counts_list[-2] != 0:
+            multiallelic_partitions = 1
+        logger.info(
+            f"Number of multiallelic partions to search: %i", multiallelic_partitions
+        )
+
+    ht_filt = ht_whole._filter_partitions(
+        sample(range(ht_whole.n_partitions()), multiallelic_partitions)
+    )
+    group_ht = ht_filt.group_by(ht_filt.locus).aggregate(n=hl.agg.count())
+    group_ht = group_ht.filter(group_ht.n > 5)
+    n_multiallelic = group_ht.count()
+
+    # Collect all variants at highly multiallelic loci
+    # Do collect() and filter() since it is fairly easy just for one field (locus) , instead of an inner join
+    multi_list = group_ht.locus.collect(_localize=False)
+    ht_multiallelic = ht_filt.filter(multi_list.contains(ht_filt.locus))
+    logger.info(
+        f"There are %i very multiallelic sites with %i total variants that will be added",
+        n_multiallelic,
+        ht_multiallelic.count(),
+    )
+
+    # Checkpoint for final multiallelic variants
+    ht_multiallelic = ht_multiallelic.checkpoint(
+        f"{args.tmp_dir}/multiallelic_path_temp.ht",
+        _read_if_exists=args.read_if_exists,
+        overwrite=args.overwrite,
+    )
+
+    # Assemble a table of variants altered by Minimum Representation
+    # NOTE: Construct Table only by partition, assume each partition has ~600 minrep variants
```

Comment:
where did this 600 come from?

Reply:
Previously we had 3000 min-rep variants from 5 partitions , so an average of 600 variants from each partition